tls/s2n_connection.c

```diff
@@ -427,6 +426,24 @@ int s2n_connection_set_config(struct s2n_connection *conn, struct s2n_config *co
 {
     notnull_check(conn);
     notnull_check(config);
+
+    if(s2n_x509_trust_store_has_certs(&config->trust_store)) {
+
+        s2n_x509_validator_init(&conn->x509_validator, &config->trust_store, config->check_ocsp);
+
+        if(!conn->verify_host_fn) {
+            conn->verify_host_fn = config->verify_host;
+            conn->data_for_verify_host = config->data_for_verify_host;
+        }
+    }
+    else {
+        if(config->client_cert_auth_type == S2N_CERT_AUTH_REQUIRED) {
+            return -1;
+        }
+
+        s2n_x509_validator_init_no_checks(&conn->x509_validator);
```

Comment:
I think it would be safer to make skipping x509 Validation it's own API call, rather than implicitly happen if you try setting a s2n_config with an empty Trust Store.

Reply:
I'll play around with this and see what I can come up with.

Reply:
ok, two updates in this regard. First, we now default to always validating AND we automatically setup the keystore to use the default location for the host operating system. Second, I added a function to explicitly disable validation.